src/manifold/src/boolean_result.cpp

```diff
@@ -772,6 +772,7 @@ Manifold::Impl Boolean3::Result(OpType op) const {
 
   Vec<TriRef> refPQ = UpdateReference(outR, inP_, inQ_, invertQ);
 
+  // This causes some verts to be outside of their refPQ triangle.
   outR.SimplifyTopology();
```

Comment:
I tried just reversing the order - create the properties first, and then simplify the topology. However, in this case it's very difficult to keep from duplicating prop verts (CoplanarProp test does a good job catching this). This might not seem like a huge deal, but they tend to continue to multiply, until your final output going to the renderer is basically triangle soup, even if the geometry is manifold.